zerver/models.py

```diff
@@ -2150,6 +2150,16 @@ def status_from_string(status: str) -> Optional[int]:
 
         return status_val
 
+class UserStatus(models.Model):
+    user_profile = models.ForeignKey(UserProfile, on_delete=CASCADE)  # type: UserProfile
```

Comment:
Should this has `unique=True`?  We definitely want an index in any case, and as we discussed, the plan is to just have a singular status for the user.